src/libpmemfile/preload.c

```diff
@@ -2721,16 +2721,133 @@ establish_mount_points(const char *config)
 	if (getcwd(cwd, sizeof(cwd)) == NULL)
 		exit_with_msg(PMEMFILE_PRELOAD_EXIT_GETCWD_FAILED, "!getcwd");
 
-	struct stat kernel_cwd_stat;
-	if (stat(cwd, &kernel_cwd_stat) != 0) {
+	if (stat(cwd, kernel_cwd_stat) != 0) {
 		exit_with_msg(PMEMFILE_PRELOAD_EXIT_CWD_STAT_FAILED,
 				"!fstat cwd");
 	}
+}
 
-	assert(pool_count == 0);
+static void
+init_pool(struct pool_description *pool_desc, struct stat *kernel_cwd_stat)
+{
+	/* fetch pool_desc-fd, pool_desc->stat */
+	open_mount_point(pool_desc);
+
+	pool_desc->pool = NULL;
+
+	util_mutex_init(&pool_desc->pool_open_lock);
+	util_mutex_init(&pool_desc->process_switching_lock);
+
+	++pool_count;
+
+	/*
+	 * If the current working directory is a mount point, then
+	 * the corresponding pmemfile pool must opened at startup.
+	 * Normally, a pool is only opened the first time it is
+	 * accessed, but without doing this, the first access would
+	 * never be noticed.
+	 */
+	if (same_inode(&pool_desc->stat, kernel_cwd_stat)) {
+		open_new_pool(pool_desc);
+		if (pool_desc->pool == NULL) {
+			exit_with_msg(PMEMFILE_PRELOAD_EXIT_POOL_OPEN_FAILED,
+				"!opening pmemfile_pool");
+		}
+		cwd_pool = pool_desc;
+	}
+}
+
+static void
+detect_mount_points(struct stat *kernel_cwd_stat)
+{
+	FILE *file = fopen("/proc/self/mountinfo", "r");
 
+	if (!file)
+		return;
+
+	unsigned mount_id, parent_id, major, minor;
+	char root[PATH_MAX];
+	char mount_point[PATH_MAX];
+	char mount_options[4096];
+	char f[9][4096];
+	int matched = 0;
+	size_t len = PATH_MAX;
+	char *line = malloc(len);
+	if (!line) {
+		fclose(file);
+		return;
+	}
+
+	while (getline(&line, &len, file) > 0) {
+		matched = sscanf(line,
+			"%u %u %u:%u %s %s %s %[^\n ] %[^\n ] %[^\n ] %[^\n ] %[^\n ] %[^\n ] %[^\n ] %[^\n ] %[^\n ]",
+			&mount_id, &parent_id, &major, &minor, root,
+			mount_point, mount_options, f[0], f[1], f[2], f[3],
+			f[4], f[5], f[6], f[7], f[8]);
+
+		if (matched <= 0)
+			break;
+
+		int i;
+		for (i = 7; i < matched; ++i) {
+			if (strcmp(f[i - 7], "-") == 0) {
+				i++;
+				break;
+			}
+		}
+		if (i == matched)
+			continue;
+		const char *fstype = f[i - 7];
+		const char *mount_source = f[i - 7 + 1];
+
+		if (strcmp(fstype, "tmpfs") != 0)
+			continue;
+		if (strncmp(mount_source, "pmemfile:", 9) != 0)
```

Comment:
Instead of hard-coded written `9` in several places:
```
static const char prefix[] = "pmemfile:";

if (strncmp(mount_source, prefix, strlen(prefix)) != 0)
      continue;

mount_source += strlen(prefix);
// and now mount_source points to the first char of the path
```

Reply:
Done.